sycl/include/sycl/ext/intel/experimental/esimd/simd_view.hpp

```diff
@@ -41,6 +44,9 @@ class simd_view : public detail::simd_view_impl<BaseTy, RegionTy,
   /// The simd type if reading this simd_view object.
   using value_type = simd<element_type, length>;
 
+  // Construct a complete view of a vector
+  simd_view(BaseTy &Base) : BaseClass(Base) {}
```

Comment:
Do length 1 specializations support this form of constructor? I believe they should as well.

Reply:
Currently no. If I add this constructor, then we still cannot write:
```
simd<int, 1> s = 0;
simd_view v(s); // error
simd_view<simd<int, 1>, int, 1, 1> v(s); // OK
```
Since that will require having default arguments in the length 1 specialization classes. And default template argument in a class template partial specialization are not allowed([link](https://stackoverflow.com/questions/18700558/default-template-parameter-partial-specialization)).

I haven't yet found a way how we could allow this for length 1 specializations. We may still need it in the future, but for now, the w/a is simple: `auto v = s[0]`.